dataPreserve.js

```diff
@@ -0,0 +1,59 @@
+function storageAvailable(type) {
+  let storage;
+  try {
+    storage = window[type];
+    const x = '__storage_test__';
+    storage.setItem(x, x);
+    storage.removeItem(x);
+    return true;
+  } catch (e) {
+    return e instanceof DOMException && (
+      e.code === 22
+
+            || e.code === 1014
+
+            || e.name === 'QuotaExceededError'
+
+            || e.name === 'NS_ERROR_DOM_QUOTA_REACHED')
+
+            && (storage && storage.length !== 0);
+  }
+}
+
+if (storageAvailable('localStorage')) {
+  const formData = {
+    name: '',
+    email: '',
+    comment: '',
+  };
+  const formName = document.getElementById('name-field');
+  const formEmail = document.getElementById('email-form');
+  const formComment = document.getElementById('message');
+  formName.addEventListener('input', () => {
+    formData.name = formName.value;
+    localStorage.setItem('formDataItem', JSON.stringify(formData));
+  });
+
+  formEmail.addEventListener('input', () => {
+    formData.email = formEmail.value;
+    localStorage.setItem('formDataItem', JSON.stringify(formData));
+  });
+  formComment.addEventListener('input', () => {
+    formData.comment = formComment.value;
+    localStorage.setItem('formDataItem', JSON.stringify(formData));
+  });
+
+  if (localStorage.getItem('formDataItem')) {
+    const dataFromStorage = JSON.parse(localStorage.getItem('formDataItem'));
+    formName.value = dataFromStorage.name;
+    formEmail.value = dataFromStorage.email;
+    formComment.value = dataFromStorage.comment;
+  } else {
+    formData.name = formName.value;
+    formData.email = formEmail.value;
+    formComment.comment = formComment.value;
+    localStorage.setItem('formDataItem', JSON.stringify(formData));
+  }
+} else {
+  alert('plaase enable the local storage to persist data within the form');
```

Comment:
- [ ] Kindly avoid using alert, it's not good practice as sometimes it stops code execution and displays differently on different browsers. Use a custom modal instead.